File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        charset = set()
        max_len = 0
        l = 0
        
        for r in range(len(s)):
            while s[r] in charset:
                charset.remove(s[l])
                l += 1
           
            charset.add(s[r])
            max_len = max(r - l + 1,max_len)
            
        return max_len
                
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        var sets= mutableSetOf<Char>()
        
        var l = 0
        var ans = 0
        for(i in 0.. s.length-1 ){
            while(sets.contains( s[i] ) ){
                sets.remove(s[l])
                 l += 1
            }
            sets.add(s[i])
            ans = maxOf(ans,i-l+1)
            
        }
        return ans
        
    }
}